src/resources/js/dropdowns.js

```diff
@@ -83,38 +83,76 @@ var tribe_dropdowns = tribe_dropdowns || {};
 		return result;
 	};
 
-	obj.init_selection = function ( element, callback ) {
-		var $select = element,
-			args = $select.data( 'dropdown' ),
-			data = [],
-			is_multiple = $select.is( '[data-multiple]' );
-
-		$( element.val().split( args.regexSplit ) ).each( function () {
-			var item = { id: this, text: this };
-			if ( args.data.length > 0  && _.isObject( args.data[0] ) ) {
-				var _item = _.where( args.data, { text: this.valueOf() } );
-				if ( _item.length > 0 ){
-					item = _item[0];
-				} else {
-					_item = _.where( args.data, { id: parseInt( this.valueOf(), 10 ) } );
-					if ( _item.length > 0 ){
-						item = _item[0];
-					}
-				}
-			}
-
-			if ( is_multiple ) {
-				data.push( item );
-			} else {
-				callback( item );
+	/**
+	 * If the element used as the basis of a dropdown specifies one or more numeric/text
+	 * identifiers in its val attribute, then use those to preselect the appropriate options.
+	 *
+	 * @param {object}   $select
+	 * @param {function} make_selection
+	 */
+	obj.init_selection = function( $select, make_selection ) {
+		var is_multiple    = $select.is( '[data-multiple]' ),
+		    options        = $select.data( 'dropdown' ),
+		    current_values = $select.val().split( options.regexSplit ),
+		    selected_items = [];
+
+		$( current_values ).each( function() {
+			var search_for   = { id: this, text: this };
+			var located_item = find_item( search_for, options.data  );
+
+			if ( located_item ) {
+				selected_items.push( located_item );
 			}
 		} );
 
-		if ( is_multiple ) {
-			callback( data );
+		if ( selected_items.length && is_multiple ) {
+			make_selection( selected_items );
+		} else if ( selected_items.length ) {
+			make_selection( selected_items[ 0 ] );
 		}
 	};
 
+	/**
+	 * Searches array 'haystack' for objects that match 'description'.
+	 *
+	 * The 'description' object should take the form { id: number, text: string }. The first
+	 * object within the haystack that matches one of those two properties will be returned.
+	 *
+	 * If objects contain an array named 'children', then that array will also be searched.
+	 *
+	 * @param {Object} description
+	 * @param {Array}  haystack
+	 *
+	 * @return {Object|boolean}
+	 */
+	function find_item( description, haystack ) {
```

Comment:
Hmm, well definitely two different ways we can go here; my own feeling is we should avoid a large public surface area, same as in our server-side code where we default to private/protected levels of visibility.

Also, if we see it as a good general purpose function, maybe the dropdown lib isn't the ideal home for it anyway - else non-dropdown-specific code needs to include this to utilize the function).

It could possibly go in `tribe-common.js` if we wanted to do that, but I'm not sure a) if we need to do that now b) if it will make sense following resolution of your other note re using a reducer.